File: src/gestortaller/Bus.java
package gestortaller;
import gestorviajes.*;
public class Bus extends Terrestre {
	private final static int NUM_SILLAS_ECONOMICAS = 28;
	private final static int NUM_SILLAS_EJECUTIVAS = 12;

	public Bus(String nombre, Transportadora transportadora) {
		super(nombre, transportadora);
		this.setSILLASECONOMICAS(new Silla[NUM_SILLAS_ECONOMICAS]);
		this.setSILLASEJECUTIVAS(new Silla[NUM_SILLAS_EJECUTIVAS]);
		Ubicacion ubicacion = null;

		for (int numPosicion = 0; numPosicion < NUM_SILLAS_EJECUTIVAS; numPosicion++) {
			if (numPosicion % 4 == 0 || numPosicion % 4 == 3) {
				ubicacion = Ubicacion.VENTANA;
			} else {
				ubicacion = Ubicacion.PASILLO;
			}

			this.getSILLASEJECUTIVAS()[numPosicion] = new Silla(Clase.EJECUTIVA, numPosicion, ubicacion);
		}
		for (int numPosicion = 0; numPosicion < NUM_SILLAS_ECONOMICAS; numPosicion++) {
			if (numPosicion % 6 == 0 || numPosicion % 6 == 5) {
				ubicacion = Ubicacion.VENTANA;
			} else if (numPosicion % 6 == 1 || numPosicion % 6 == 4) {
				ubicacion = Ubicacion.CENTRAL;
			} else if (numPosicion % 6 == 2 || numPosicion % 6 == 3) {
				ubicacion = Ubicacion.PASILLO;
			}
			this.getSILLASECONOMICAS()[numPosicion] = new Silla(Clase.ECONOMICA, numPosicion, ubicacion);
		}
	}
	@Override
	public String toString() {
		return this.getNombre() + "_A";
	}

	public static int getNumSillasEconomicas() {
		return NUM_SILLAS_ECONOMICAS;
	}

	public static int getNumSillasEjecutivas() {
		return NUM_SILLAS_EJECUTIVAS;
	}
	public double Calcular_Consumo_Gasolina(double distancia_en_km) {
		double consumido;
		consumido = this.getGastoGasolina() * distancia_en_km;
		return consumido;
	}
}

File: practica02/Practica2VALE/Practica02VALE/SRC/gestorAplicacion/parada/Bus.py
import math
from gestorAplicacion.parada.Terrestre import Terrestre
from gestorAplicacion.parada.Silla import Silla
from gestorAplicacion.parada.Clase import Clase
from gestorAplicacion.parada.Ubicacion import Ubicacion


class Bus(Terrestre):
    _NUM_SILLAS_ECONOMICAS = 32
    _NUM_SILLAS_EJECUTIVAS = 12

  
    def __init__(self, nombre, transportadora):
        super().__init__(nombre, transportadora)
       
        ubicacion = None

     
        for numPosicion in range(0,Bus._NUM_SILLAS_EJECUTIVAS):
            if numPosicion%4 == 0 or numPosicion % 4 == 3:
                ubicacion = Ubicacion.VENTANA
            else:
                ubicacion = Ubicacion.PASILLO

            self.getSILLASEJECUTIVAS().append( Silla(Clase.EJECUTIVA, numPosicion, ubicacion))

       
        for numPosicion in range(0, Bus._NUM_SILLAS_ECONOMICAS):
            if numPosicion % 6 == 0 or numPosicion% 6 == 5:
                ubicacion = Ubicacion.VENTANA
            elif numPosicion % 6 == 1 or numPosicion % 6 == 4:
                ubicacion = Ubicacion.CENTRAL
            elif numPosicion % 6 == 2 or numPosicion % 6 == 3:
                ubicacion = Ubicacion.PASILLO
            self.getSILLASECONOMICAS().append(Silla(Clase.ECONOMICA, numPosicion, ubicacion))

    def getNombre(self):
        texto = super().getNombre() + "_V"
        return texto

    @staticmethod
    def getNumSillasEconomicas():
        return Bus._NUM_SILLAS_ECONOMICAS

    @staticmethod
    def getNumSillasEjecutivas():
        return Bus._NUM_SILLAS_EJECUTIVAS

    # METODOS

    def Calcular_Consumo_Gasolina(self, distancia_en_km):
        consumido = None
        consumido = self.getGastoGasolina() * distancia_en_km
        return consumido
